src/lib/native.ts

```diff
@@ -86,3 +88,26 @@ export async function driveStats(dir: string): Promise<DriveStats> {
   const stats: DriveStats = { freeBytes: result.free_bytes, totalBytes: result.total_bytes }
   return stats
 }
+
+export async function winregGet(subKey: string, value: string): Promise<string> {
+  const result = (await tauri.invoke('winreg_get', { subKey, value })) as string
+  console.log(result);
+  return result
+}
+
+export async function winregSet(subKey: string, setKey: string, value: string): Promise<string> {
+  const result = (await tauri.invoke('winreg_set', { subKey, setKey, value })) as string
+  console.log(result);
+  return result
+}
+
+export async function winregDelete(subKey: string, setKey: string): Promise<string> {
+  const result = (await tauri.invoke('winreg_delete', { subKey, setKey })) as string
+  console.log(result);
+  return result
+}
```

Comment:
These should really be called differently all the way through. The fact that implementation detail (library `winreg`) leaks all the way here is not pretty.

Also is `console.log` really needed here?

Reply:
How would you suggest to structure these functions? These are mostly just a wrappers to add type checking around the tauri.invoke functions that call my rust functions.

Reply:
Both these functions and those that are in Rust shouldn't have `winreg` prefix.

For instance, instead of `function winregGet(subKey: string, value: string): Promise<string>` I would expect `function isWindowsAutostartEnabled(): Promise<boolean>` (returns boolean if everything is fine and throws `Error` if something went wrong).

Reply:
The Rust functions and ts wrappers are meant to be general windows registry utility functions, they can be used to read/write any registry data, which will be useful for more windows functionality in the future. I can refactor into special case functions but imo there is no benefit to doing so.

Reply:
Having direct access from frontend into Windows Registry or being able to run arbitrary executables is a security issue waiting to happen, also it makes it nearly impossible to audit what frontend can or can't do. Clear purpose-built and restricted methods are easier to use, easier to audit and easier to keep secure.

Reply:
Your comment also applies to the applescript utility functions so I made a TODO to refactor winreg and applescript related functions to make them more use case specific. I also have functions that can read/write files to arbitrary directories that could be refactored.